### ISSUE TITLE
design issue: fetch handler doesn't provide response on error

### ISSUE THREAD

**Issue by BryanCrotaz** [emberjs/data#9133]
Hitting the api and receiving an error (eg 401) is hard to parse with the current design:

```
// my-handler.ts
...
try {
  return await next(request);
catch (e) {
  // e is an Error new Error(`[${response.status}] ${response.statusText} - ${response.url}`)
  if (e.message.contains('401')) {
    // redirect to login
  }
}
```

If error contained the response object I could write
```
// my-handler.ts
...
try {
  return await next(request);
catch (e) {
  if (e.response.status == 401) {
    // way more readable, and more performant
    // redirect to login
  }
}
```

## Suggestion
Attach the response object to the Error that's thrown.

**Comment by runspired** [comment]
We haven't backported all the changes we made to the Fetch handler, but here's what it does have today: https://github.com/emberjs/data/blob/5a8560a1a7190456e5a03938718c2461edb6714a/packages/request/src/fetch.ts#L144-L153

We've already listed all improvements to the request service as things to backport to 5.3 and 4.12, so closing this.

**Comment by BryanCrotaz** [comment]
Which version do I need to be using to get these improvements?

**Comment by runspired** [comment]
@BryanCrotaz currently its just in canary (maybe, canary hasn't auto published in a while because the release script needs adjusted for recent work, which is something I'm working on). We plan to put those improvements into both 4.12 and 5.3 LTS
